APE 24: Astropy Affiliated Packages with pyOpenSci

pllim opened the pull request:
Initial authors: @pllim @lwasser @WilliamJamieson @dhomeier

* Close https://github.com/astropy/astropy-project/issues/350
* Close https://github.com/astropy/astropy-project/issues/340
* Close https://github.com/astropy/astropy-project/issues/139
* Fix https://github.com/astropy/astropy-project/issues/334

### TODO

- [x] Start a draft APE during Scipy 2023.
- [x] Finish filling in the text.
- [x] Have co-authors proof-read and address feedback.
    - ✔️ Leah
    - ✔️ Derek
    - ✔️ William
    - ✔️ Moritz
- [x] Open up to general comment period and address feedback. Done on 2023-10-23
- [x] Address feedback from general comment period.
- [x] Wait for CoCo decision. 2023-12-18: 2 abstained, 2 approved, 1 absent
- [ ] CoCo to update PR with decision and merge.
- ???
- Profit!!!

pllim commented:
@lwasser @WilliamJamieson @dhomeier -- Please proof-read before I open this up to the rest of Astropy. Thank you!

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+
+* Larger exposure for Affiliated packages within scientific Python ecosystem,
+  not just Astropy.
+* They get to be listed in JOSS "for free" from the
```
Can we be a bit careful with this language. I'd suggest something like

If there are in scope for JOSS, they can be fast-tracked through the JOSS review process and get a publication. 

This is important because there are some packages that JOSS won't accept given scope but we would accept. And

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+Who owns this process?
+----------------------
+
+pyOpenSci already has an ongoing process and it is a community-driven one.
+It is compatible with Astropy but it is up to us to decide if a package
+applied to pyOpenSci also qualifies as Affiliated.
```
```suggestion
pyOpenSci already has an established software peer review process and it is a community-driven.
It is compatible with Astropy but it is up to us to decide if a package
applied to pyOpenSci also qualifies as Affiliated.
```

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+pyOpenSci has its own funding structure beyond the control of Astropy.
+As of July 2023, it is fully funded for the next 2 years.
+Given the number of projects currently looking into a similar partnership
+with pyOpenSci (SunPy, Heliophysics, etc.) and that it has a full-time staff
+dedicated to seeking out more funding, we find that the benefits outweigh
```
```suggestion
with pyOpenSci (SunPy, PlasmaPy, pyHeliophysics etc.) and that it has a full-time staff
dedicated to seeking out more funding, we find that the benefits outweigh
```

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+`submission guide for package authors <https://www.pyopensci.org/software-peer-review/how-to/author-guide.html>`_,
+`accompanying GitHub issue templates <https://github.com/pyOpenSci/software-submission/issues/new/choose>`_, and
+`listing of accepted packages <https://www.pyopensci.org/python-packages.html#our-accepted-python-open-source-packages>`_.
+As of July 2023, they are in the process of revamping the submission form to handle
```
yes so  i've added the peer review link to our packages page already!

<img width="1276" alt="Screen Shot 2023-08-01 at 4 19 58 PM" src="https://github.com/astropy/astropy-APEs/assets/7649194/601f4c59-d0a7-4708-ad94-46c7e30098c1">

but we might want to reword this to not conflate the website (image above) with our submission form (a github template issue).

pllim replied:
I will update the text to reflect that you have completed the link to reviews (👏) but I am not sure what you mean by "reword this to not conflate the website (...) with our submission form". I did link to your template too, so hope you can clarify. Thanks!

pllim replied:
@lwasser , can you please clarify on what else I need to change to satisfy "reword this to not conflate the website (...) with our submission form"?

lwasser replied:
@pllim i think we can resolve this. some of me wondered if this was attached to the wrong place or text is gone. i don't see any reference to linking to the reviews on the website in this section of the content. I think originally i must have seem some language around linking to the review from the website mixed in to the review process itself. and those two things are separate. so let's just resolve this comment.

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+`accompanying GitHub issue templates <https://github.com/pyOpenSci/software-submission/issues/new/choose>`_, and
+`listing of accepted packages <https://www.pyopensci.org/python-packages.html#our-accepted-python-open-source-packages>`_.
+As of July 2023, they are in the process of revamping the submission form to handle
+collaborations with different communities (e.g., Astropy), including but not limited to:
+
+* creating a community-specific label for filtering and link to current reviews
+  for that community; and
+* customizing the editorial bot to add community-specific tags.
```
```suggestion
collaborations with different communities (e.g., Astropy). There are also:

* creating a community-specific label for package filtering and link to current reviews
  for that community on their website package listing; and
* customizing the editorial bot to add community-specific tags.
```

lwasser replied:
i'm not sure if this is quite right but

there are two buckets of activities here

1. updating the website to display packages and have astropy filtering 
2. updating the submission template so a potentially astropy affiliated package can select that option for the actual review. 
3. we can totally add an astropy label to github - that is easy

we don't have a bot YET - BUT i'll be working on that with Arfon from JOSS starting in the late fall to automate some of our review steps :) Right now the editors would do this :)

pllim replied:
I updated the bullet points. I hope they reflect all your comments here.

lwasser replied:
this looks good!

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+because we modeled the process after traditional astronomical journals.
+Even then, the reviewer had the option to reveal themselves if they wanted to.
+Given that the Astropy ecosystem is open-source anyway, it makes sense for us
+to move away from the journal-style tradition and embrace a more open process that
```
yay for open everything!!

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+it does *not* have to come from the Astropy community directly and will be objective.
+Rejection can also happen if the maintainer of the package under review
+is unwilling to ensure the package can be used and maintained.
+While in practice, rejection is rare, usually the Editor-in-Chief would be
```
Can we adjust this just a bit. rejection could happen at varying stages BUT it's most likely it would happen during the pre-review checks that the EIC implements. so you are right that most often the EIC would perform that step. however please also know it could be possible for a package to begin the review process and be rejected as well. and that rejection would come from the editor for that review. 

we haven't had that happen yet. but it could happen. 
Alternatively what can also happen is a package maintainer might stop responding on a review. so we'd close it. i just want to ensure this is clear here but i totally understand what you are saying!

pllim replied:
Okay, I will rewrite this.

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+While in practice, rejection is rare, usually the Editor-in-Chief would be
+the one doing the rejection.
+
+A package may still be listed under pyOpenSci (and JOSS) if it
+passes the general criteria but fails the Astropy-specific ones;
+e.g., a very nice Python package that has little to do with the Astropy ecosystem.
+
```
```suggestion
A package that fails the Astropy-specific criteria may still be accepted into the pyOpenSci ecosystem and also published by JOSS) if it is in scope;
an example of this might be a well-documented and develops Python package that is useful to the scientific community that fails to relate well to the Astropy ecosystem.

```

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+However, it has plans to set up automated checks that track the "health"
+and maintenance level of a package over time, with the goal of identifying
+packages that have become "orphans" (i.e, unmaintained). Once a package is
+identified as needing additional maintenance, it will be flagged for the community.
```
```suggestion
identified as needing additional maintenance, it will be flagged.
```

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+Some, if not most, of this data will be collected using
+`devstats <https://github.com/scientific-python/devstats>`_ and
+`repo-review <https://github.com/scientific-python/repo-review>`_ from
+`Scientific Python`_. This information will be displayed on public dashboards.
```
```suggestion
`Scientific Python`_. This information will be displayed on public dashboards hosted on the pyOpenSci website.
```

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+`repo-review <https://github.com/scientific-python/repo-review>`_ from
+`Scientific Python`_. This information will be displayed on public dashboards.
+
+However, automation can only go so far; for instance, if automation
```
Just a note - i suspect we can figure this out. i believe repo_review can handle gitlab. BUT i like this text as a caveat!

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+
+If the package is no longer maintained, one of the following can happen:
+
+1. If it is a widely used package and the maintainer wants to see it live on,
+   they will try to help the maintainer to find a new maintainer team;
+2. Otherwise, they will gracefully sunset it from the list of maintained/accepted
+   packages.
```
```suggestion
1. If it is a widely used package and the maintainer wants to see it live on,
   pyOpenSci will try to help the maintainer build a new maintainer team;
2. Otherwise, pyOpenSci will gracefully sunset it from the list of maintained/accepted
   packages.
```

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+listed once more. An example scenario that might happen is when the sole
+maintainer goes on a hiatus longer than the time frame set in
+`maintainer responsiveness <https://www.pyopensci.org/software-peer-review/our-process/policies.html#maintainer-responsiveness>`_
+and then comes back to a sunsetted package.
```
excellent!

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+This guide has a stringent community review process. Reviewers for this guide
+represent members of PyPA, core Python, Anaconda (conda/conda-forge), and
+core packages for front- and back-end tools (e.g., ``flit``, PDM, ``hatch``/``hatchling``, etc.).
+It is also maintained by `Scientific Python`_.
```
hmmm maybe you are referencing a different guide? we maintain that guide at pyOpenSci but henry's scikit-hep guide is on the scientific python website now. that is a different guide with a very different development process

pllim replied:
I was referencing to https://www.pyopensci.org/python-package-guide/ . I saw some Scientific Python faces under https://github.com/pyopensci/python-package-guide#contributors- . But you are saying that your guide has nothing to do with Scientific Python? Should I remove this text then?

```rst
It is also maintained by `Scientific Python`_.
```

pllim replied:
Should I also mention https://learn.scientific-python.org/development/guides/packaging-classic/ or is that not in scope? Hope you can advise. Thanks!

lwasser replied:
hey @pllim some of the scientific python project people absolutely contributed to our guide via our community review process!! but our guide on the pyOpenSci website is maintained by the pyOpenSci  community not the scientific Python project. i know it's confusing but they are separate entities.  The other guide https://learn.scientific-python.org/development/guides/packaging-classic/  is excellent but different from ours and largely developed by Henry from the scikit HEP community. so for our review process and the APE around peer review we probably want to  stick to our guide. but i think both guides are useful so if you want to link to the other guide in your documentation around packaging that is wonderful.

i hope that helps. we just wont consider that other guide a part of our peer review process as it isn't a pyos community   entity that we contribute to or maintain.

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+pyOpenSci is willing to create the following specifically for Astropy
+if we agree to this partnership:
+
+* A RSS/XML feed of Affiliated packages that we could ingest and
```
do we want to specify the format here? we could do xml! but i already also have a yaml file that you could grab right now. so i just wonder rather than specifying the format, if we want to just say a automated way for astropy to access package information and publish on their website. something a bit more generic. then we can talk about what format makes the most sense.

pllim replied:
Sure. I think I got that from one of those notes we took.

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+   We would also link back to pyOpenSci on our page.
+   Meanwhile, we would keep the pre-APE 24 listing on a "legacy affiliation" page
+   that will be kept alive during the lifetime of the Astropy project.
+5. Work with pyOpenSci to cross-list Affiliated editors, between their
```
this is just an idea that i just had - but i was thining it might be cool to have a little astropy logo on the [editor listing page](https://www.pyopensci.org/about-peer-review/index.html#our-editorial-board) that would highlight astropy editors. it would also provide more visibility for astropy on multiple pages on our website. :)

pllim replied:
Added your idea to the APE. Thanks!

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,443 @@
+   that will be kept alive during the lifetime of the Astropy project.
+5. Work with pyOpenSci to cross-list Affiliated editors, between their
+   editors listing and our roles page.
+6. Work with pyOpenSci to hash out a more concrete plan on how to
```
yes this is a situation i hadn't considered yet. but we can hash that out.

lwasser commented:
ok @pllim i added some feedback here. generally i think it looks great! this was a lot of work! please let me know if you need anything else.

lwasser commented:
hey there @pllim 👋 just following up on our slack chat here! In the APE i think it would make sense to mention that the astropy editors and reviewers will need to [adhere to our timeline](https://www.pyopensci.org/software-peer-review/our-process/review-timeline.html) so the package maintainer have a good experience with the review and things don't hang for long periods of time. Generally the editor role shouldn't take a huge amount of time BUT it's important for an editor, once the review starts, to check in on the review periodically - ever few weeks or so and more when it's wrapping up. this allows the maintainer to have a good start to end experience and avoids reviews dragging out for months to a year!

lwasser commented:
@pllim i am good with this as is 🎉 !! i left a note about the two package guides - scientific python project vs pyopensci. so the main point of clarification is that we maintain our own community developed packaging guide so that is what we rely on and direct people to for peer review if they have questions  - the scientific python dev guide is a separate document that is maintained by someone else. so i probably wouldn't add it to the ape surrounding peer review here with us specifically but i think it's a great resource that the astropy community should take advantage of. 

i hope that helps. but again i support this moving forward / being merged via whatever process your community has in place! 

let me know if you need anything else from me and my apologies for missing your replies to my comments above.

Cadair commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+represent members of PyPA, core Python, Anaconda (conda/conda-forge), and
+core packages for front- and back-end tools (e.g., ``flit``, PDM, ``hatch``/``hatchling``, etc.).
+
+Meanwhile, the Astropy community prefers a more laid-back
+`OpenAstronomy packaging guide <https://packaging-guide.openastronomy.org/en/latest/>`_.
```
I am going to object to the "laid-back" wording here, or at least ask for some more specifics.

Also, the OpenAstronomy guide is much narrower in scope to the pyOpenSci one, we describe how to make a package *which looks like astropy*. Now I know this might not be the "best" or "most popular" way to build a package, but it's what we do and what most of our coordinated packages do.

I therefore am less keen on "encouraging our Affilliated packages to follow [the pyOpenSci]" guide unless we are going to do the work of making changes to all our packages as well. (specifically here I am referring to src/ layout, but a few other things as well). Given the amount of work this would be for the debatable returns I can't see it ever happening.

pllim replied:
@lwasser , do you wish to respond to this? 🙏

dhomeier replied:
> I therefore am less keen on "encouraging our Affilliated packages to follow [the pyOpenSci]" guide unless we are going to do the work of making changes to all our packages as well. (specifically here I am referring to src/ layout,

Yes, this has met considerable resistance in astropy/astropy#13678 and I'd rather not reopen that lightly again.

lwasser replied:
i hear you. and am with you 100% . we went through the same process. matplotlib's src layout is actually for (i think) C / compiled code if i recall correctly so it's technically not even a "traditional" src/ layout. there are some packages out there using src/ now BUT people will do what they wish to do here. and it's not in any way a roadblock for this collaboration or for maintainers in general!  we just make recommendations and have accepted packages with both flat and src layouts. sounds like we are all on the same page here! 

for more specific reference [this is our criteria](https://www.pyopensci.org/software-peer-review/how-to/author-guide.html#does-your-package-meet-packaging-requirements) that our editor in chief looks for upon initial submission. because this text:

**Issue Submission Documentation** 

confuses people (i need to clarify that text!) it refers to the yaml in our review issue being filled out completed nothing to do with the package itself :) 
when we initially compared our checks to astropy's they seemed fairly well aligned. 
i really appreciate this conversation!

dhomeier replied:
> i hear you. and am with you 100% . we went through the same process. matplotlib's src layout is actually for (i think) C / compiled code if i recall correctly so it's technically not even a "traditional" src/ layout.

It's using `src` for to-be-compiled code and `lib` for the Python source, but the effect is the same, the package not being directly visible at the top-level directory.
> 
> for more specific reference [this is our criteria](https://www.pyopensci.org/software-peer-review/how-to/author-guide.html#does-your-package-meet-packaging-requirements) that our editor in chief looks for upon initial submission. because this text:

That seemed to have some potential for confusion to me as well, as those are just the entry-point criteria, but we may want to also indicate what criteria the referees (and finally the editor again) should use for accepting a package.

lwasser replied:
hey @dhomeier the best place to find all of this information is our peer review guide.

Yes, the Editor in Chief checks serve as a base level of what a package needs to begin review. These checks are similar to those in the astropy review process. We will ask an author to add documentation, etc if that doesn't exist prior to a review beginning. 

* here is the [editor guide](https://www.pyopensci.org/software-peer-review/how-to/editors-guide.html)
* and here is the [reviewer guide](https://www.pyopensci.org/software-peer-review/how-to/reviewer-guide.html#peer-review-template) that contains the template information that reviewers look for and sign off on. 

There is a guide in that chapter for all roles involved of our review process so i suggest that y'all have a look at all of the guides - maintainer / author, editor, editor in chief and reviewer to understand the big picture of our process. 

please let me know if you have any questions.

pllim replied:
On the pyOpenSci side, is it possible to have the Astropy check override the packaging guideline requirements? After all, we did agree on adding some extra domain-specific stuff.

So, I can see https://www.pyopensci.org/software-peer-review/how-to/author-guide.html#does-your-package-meet-packaging-requirements have slightly qualifiers for different domains. But if someone submit something that is cross-domain, then I think it is reasonable to require pyOpenSci standards over Astropy's.

pllim replied:
2023-11-10 -- I accepted Leah's edits about packaging and then some more. Hopefully acceptable by everyone here?

lwasser replied:
i will read through this doc in it's entirety one day this week but as far as i'm concerned astropy doesn't have to follow our packaging guidelines. They are recommendations. Things like src/ vs flat layout won't make or break the functionality of usability of a package as far as i'm concerned especially if there are tests and people know to run tests on the installed version of the package. I think src/ will generally make it a bit easier to have a solid test setup for beginners. but that is my opinion.  
We would never reject a package given it has a particular file layout. 

I do have a question for all of you here: is it the case that if someone came to you / us that is astropy related (so a good fit) but was using a src/ layout - would you reject the package or ask them to move back to a flat layout?

i just personally think that - that particularly issue is not an issue we'd ever want to reject a package over. Nor would we want someone who is following the pypa tutorials or setuptools docs and using src/ layout to be forced back to a flat layout. that is my only concern. 

i don't want to debate each layout. i just want to understand the principles here in a review associated with following openastronomy guidelines.

lwasser replied:
ok reading the comments below - https://github.com/astropy/astropy-APEs/pull/87#discussion_r1368832543 it seems like you all are on board with recommending the openastronomy guidelines but wouldn't reject a package because of a specific packaging choice that was totally valid. if that is the case let's resolve this as well.

lwasser replied:
note: we have no requirements related to the layout of the package. I think you can look at our editor in chief checks below and potentially link to it. We might suggest changes if a package has a hard-to-maintain infrastructure but it would never be rejected because of a specific layout (ie src vs flat or using a specific tool - hatch vs pdm or backend etc)

https://www.pyopensci.org/software-peer-review/how-to/editor-in-chief-guide.html#editor-checklist-template

pllim commented:
I have pinged the Editors multiple times. I think at least one of them are okay with reviewing this as part of the "public comments" period, so I took this out of draft. I will send out memo to astropy-dev.

lwasser commented:
please ping me on any questions, concerns etc! happy to help here as much as i can.

mhvk reviewed (commented):
This is excellent! I only think we should we should stick to our own packaging guide (but not insist on it).

Another thing that perhaps is worth stressing a bit is the link to JOSS - it would really help to have a simple process that also leads to a citable publication.

lwasser commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+.. _An Overview Of the Peer Review Process: https://www.pyopensci.org/software-peer-review/our-process/review-timeline.html
+
+
+Packaging guidelines
+--------------------
+
+pyOpenSci has been developing a `community-driven packaging guide`_
+that covers modern best practices and recommendations for scientific Python packaging.
+This guide has a stringent community review process. Reviewers for this guide
+represent members of PyPA, core Python, Anaconda (conda/conda-forge), and
+core packages for front- and back-end tools (e.g., ``flit``, PDM, ``hatch``/``hatchling``, etc.).
+
+Meanwhile, the Astropy community prefers a more laid-back
+`OpenAstronomy packaging guide <https://packaging-guide.openastronomy.org/en/latest/>`_.
```
```suggestion
Packaging guidelines
--------------------
The Astropy community has followed packaging guidelines published in the 
`OpenAstronomy packaging guide <https://packaging-guide.openastronomy.org/en/latest/>`_ for XX years.

pyOpenSci is also developing a `community-driven packaging guide`_
that covers modern best practices and recommendations for scientific Python packaging. 
This guide has a stringent community review process. Reviewers for this guide
represent members of PyPA, core Python, Anaconda (conda/conda-forge), and
core packages for front- and back-end tools (e.g., ``flit``, PDM, ``hatch``/``hatchling``, etc.).  

If we partner with pyOpenSci, we encourage our Affiliated packages to continue to follow the OpenAstronomy packaging guide for packaging structure. However maintainers will find the pyOpenSci guide to be a useful reference given it provides community-wide guidelines and is targeted to those newer to packaging. 

```

or something to that effect. so we aren't completely deleting our guide but making sure folks know the difference?

pllim replied:
@Cadair , does this change satisfy you?

nabobalis commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+
+* Reviewers can no longer be anonymous and have to sign up via `reviewer sign-up form`_.
+  The whole review process is open.
+* Instead of static color badges, there will be a link to full pyOpenSci
+  review for that package.
```
So as we are also planning to do the same as Astropy and partner with PyOpenSci, @Cadair pointed out that badges give even a very uniformed user a good impression of the quality of the packages.

Whereas on the surface the PyOpenSci review while more complex does not offer a glance of the overall quality from the review page.

Does anyone from Astropy have thoughts on this, at a glance ability?

nabobalis replied:
Sure the badges being outdated is a problem but in a world where we could automate those, is there no attachment to them on the astropy side?

pllim replied:
> Is there no desire from the Astropy side for the simple at a glance nature the badges could provide?

There is. But not part of the initial rollout. Once we have the automation, we could find a way to grab the new badges and re-display on Astropy page.

lwasser replied:
this is a great conversation. automating such a badge is something we've been talking about. My thought here (let me know what y'all think) is that we first get our partnership process solidified. Currently, we offer a pyOpenSci badge indicating a package has been reviewed / accepted.

However, in the future we can provide a server hosted badge (rOpenSci does this) that we can change the color of via a ci build that checks x,y,z metrics. For instance, some indication of maintenance status measured by repo activity ([see pr here on archiving projects)](https://github.com/pyOpenSci/software-peer-review/pull/258). I think we should collectively - astropy, sunpy, pyhc, scverse etc come together and discuss what types of metrics are important to us and what data would be used to capture such metrics. How fragile the data are (ie a ci build badge might change locations over time whereas github use metrics for a repo will be consistent to collect). Then we can decide how to show current status. 

Let's work on this as a community and then we can find funding to implement this work / building upon and learning from any existing efforts in this space as it makes sense.

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+William Jamieson as of July 2023) and no formal/public pool of reviewers.
+Given that reviewers are anonymous by default, we are unable to provide
+a fully transparent review process (even though results are publicly posted
+and discussed, they are posted second-hand by the editors). We are also
+unable to let people publicly sign up to be reviewers, as that would break
+anonymity. Furthermore, we have no resources nor infrastructure to
```
I don't think that's true. You could sign up publicly, but the editors select privately from a list. There i no reason why "putting your name of the list" would violate anonymity. 
```suggestion
and discussed, they are posted second-hand by the editors). Furthermore, we have no resources nor infrastructure to
```

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+* If they are in scope for JOSS, they can be fast-tracked through the JOSS
+  review process and get a publication via the
+  `partnership between pyOpenSci and JOSS <https://www.pyopensci.org/software-peer-review/partners/joss.html>`_.
+* pyOpenSci's objective way of initial triage will automatically weed out
```
True, but, at least in my experience, not a problem in practice. I don't recall (when I was affilate editor) ever seeing such a package.

pllim replied:
I saw a few. If you want to know more, we can discuss in private.

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+Astropy currently does not enforce any concrete turnaround time mainly because
+we do not know who ends up reviewing a given submission nor have control
+over their schedule. Furthermore, we only have two editors who may not have
+sufficient time themselves to actively push a review forward. Therefore, a review
+could take weeks, months, or (gasp) years.
```
I think this makes it sound worse than it is. Most reviews come in within weeks (after prodding by the editors). If not, the editors need to find a different reviewer. That's parallel to any peer review process. Even "nature" has no control over the peer reviewer's time and they still say "2 weeks" turn-around. When a review lingers, it's usually because a package does not respond in a timely manner to reviewer' requests for changes. Even that's not necessarily bad (e.g. if the reviewer suggests a big feature that they decide to include as part of their normal development process), see  e.g. https://github.com/astropy/astropy.github.com/pull/491 for an example.

If we want to be "faster" we could just close those PRs and then claim that "every package is decided within 4 week". But what would we gain by doing that?

pllim replied:
I wonder if stricter initial triaging would have weeded out the ones that dragged on, but it is neither here nor there. I accepted your text changes for this, so marking as resolved. Thanks!

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+
+Astropy currently does not enforce any concrete turnaround time mainly because
+we do not know who ends up reviewing a given submission nor have control
+over their schedule. Furthermore, we only have two editors who may not have
+sufficient time themselves to actively push a review forward. Therefore, a review
```
If that's the problem ,we need more editors. But really, sending a reminder email to a package every few weeks for ~a dozen packages per year should not overwhelm 2 editors. Again, none of this is an argument against pyopensci, but this text makes it sound as if our process is so fundamentally broken and the workload to high that Astropy cannot possibly manage and I don't think that's true.

pllim replied:
I accepted your changes, so marking as resolved. Thanks!

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+Review turnaround time
+----------------------
+
+Astropy currently does not enforce any concrete turnaround time mainly because
+we do not know who ends up reviewing a given submission nor have control
+over their schedule. Furthermore, we only have two editors who may not have
+sufficient time themselves to actively push a review forward. Therefore, a review
+could take weeks, months, or (gasp) years.
```
```suggestion
Astropy currently does not enforce any concrete turnaround time. Reviews typically come within weeks, but the response time for submitted packages to address the review is very non-uniform from "within days" to "years".
```

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+   editors in and out of `pyOpenSci Editorial Board`_, since our
+   Affiliated editor assignment is not permanent.
+
+The goal is to have a migration that is not too disruptive to current process.
+While this migration could take years to complete, the turnaround time
+for any one package should be much shorter (months, perhaps).
```
```suggestion
The goal is to have a migration that is not too disruptive to current process.
Once this APE is accepted, new package submissions will go through the pyOpenSci process. Packages currently under review can choose to continue under the old process (and then be treated and listed like packages accepted as Affiliated before this APE), or be transferred to the new process and follow pyOpenSci procedures.
```

hamogu reviewed (changes requested):
I am fully in favor of joining pyOpenSci and replacing our current process with the PyOpenSci process.

I'm also convinced that we can work out any small questions (e.g. do we recommend the OpenAstronomy packaging or pyopensci or both?).

However, I'm very surprised by how negative this APE describes our current process for affiliated packages - I'm even more surprised because both our current affiliated edotors are listed as co-authors. I've left specific suggestion in places, but the general tone in this APE is to bad-mouth our existing process and I don't think that's helpful. This is an APE for changing a process, and we should fairly describe what's happening - pyOpenSci has a lot to offer without us writing this to make Astropy look bad - because I think it's not. The concept of affilated packages - that's been in place for a decade - has been very successful for the growth of our ecosystem.

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+Astropy Affiliated Packages with pyOpenSci
+******************************************
+
+author: Pey Lian Lim, Leah Wasser, William Jamieson, Derek Homeier
```
```suggestion
author: Pey Lian Lim, Leah Wasser, William Jamieson, Derek Homeier, Moritz Guenther
```

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+
+Fast forward to July 2023, there are 47 accepted Affiliated packages and
+7 undergoing the review process. We have two editors (Derek Homeier and
+William Jamieson as of July 2023) and no formal/public pool of reviewers.
```
```suggestion
William Jamieson as of July 2023) and recruit reviewers from the Astropy community.
Editors identify community members with sufficient experience in the Astropy ecosystem and practices
to review packages, usually asking reviewers with scientific domain experience related to a potentially
affiliated package. This process mirrors what is commonly done for scientific journals, where editors 
invite scientific peers of a study to be reviewers.
```

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+a fully transparent review process (even though results are publicly posted
+and discussed, they are posted second-hand by the editors). We are also
+unable to let people publicly sign up to be reviewers, as that would break
+anonymity. Furthermore, we have no resources nor infrastructure to
+re-review the 47 accepted packages to ensure that they still meet the
+Affiliated standards. Therefore, there is a need to restructure our Affiliated
```
```suggestion
anonymity. 

Unlike scientific articles, which are static after publication, software continues to develop
adding features and bugs; if it is not continuously developed it will become obsolete and might
not work with future Numpy and Python versions. Thus, a process of "re-review" for affiliated packages
has been envisioned, but never executed because 
we have no resources nor infrastructure to
re-review the 47 accepted packages to ensure that they still meet the
Affiliated standards. Therefore, there is a need to restructure our Affiliated
```

pllim replied:
This change conflicts with your other change right above. I will have to guess the original intention and apply it separately.

pllim replied:
Applied.

hamogu commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+Finding reviewers
+-----------------
+
+pyOpenSci currently has a list of reviewers who have signed up for this task using the
+`reviewer sign-up form`_.
```
@lwasser Does pyopensci ever contact people as reviewers who have not signed-up by themselves? Knowledgeable editors will often know domain experts that might make good reviewers. Many of those might not know about pyopensci or be willing to actively opt-in, but would review a package is asked.

lwasser replied:
@hamogu absolutely. in fact often that is just what happens by default because editors as you point out know people. that is why it's really helpful to have an editor with domain expertise who knows peopel in the community! 

the other thing that happens is a reviewer signs up and when they sign up they have time. Then  we need their expertise say 3-5 months later and they are busy. so the signup / list of people is useful but sometimes doesn't represent who has time when we need that specific expertise.

pllim commented:
> make Astropy look bad

That was not my intention at all and I apologize if the effect is such. I was trying to describe the current situation in what I thought an objective way to provide motivation on why we decided to revamp the process. I was also an Editor, so the original text was from my own experience, plus what I think I saw after. I was not involved before that and could not provide an objective summary of the experience retroactively.

tl;dr -- I take full responsibility of the text you found offensive and I apologize. Thanks for the edits!

pllim commented on APE24.rst:
```diff
@@ -0,0 +1,497 @@
+packages that have become "orphans" (i.e, unmaintained). Once a package is
+identified as needing additional maintenance, it will be flagged.
+
+"Health" might include frequency of commits, releases, CI status, etc.
+Some, if not most, of this data will be collected using
+`devstats <https://github.com/scientific-python/devstats>`_ and
+`repo-review <https://github.com/scientific-python/repo-review>`_ from
+`Scientific Python`_. This information will be displayed on public dashboards
+hosted on the pyOpenSci website; we may also grab those same badges,
+where applicable, for a similar dashboard on Astropy website
+focused only in our Affiliated packages.
```
@nabobalis , while this won't be immediately available, it is planned. Would such a plan make you less nervous about the current static badges going away?

nabobalis replied:
I have no personal issue with those going, I was just voicing the concerns from @Cadair. 

I think this plan is great and I support it 100%.

pllim commented:
@hamogu , does the draft look acceptable to you now? If so, please clear your "changes requested" in the review. Thanks!

hamogu commented:
> tl;dr -- I take full responsibility of the text you found offensive and I apologize. Thanks for the edits!

Offensive is too strong a word. I would prefer "respectfully disagreed".

dhomeier commented on APE24.rst:
```diff
@@ -0,0 +1,497 @@
+core packages for front- and back-end tools (e.g., ``flit``, PDM, ``hatch``/``hatchling``, etc.).  
+
+If we partner with pyOpenSci, we encourage our Affiliated packages to continue to follow the
+`OpenAstronomy packaging guide`_ for packaging structure. However maintainers will find the
+pyOpenSci guide to be a useful reference, given it provides community-wide guidelines and
+is targeted to those newer to packaging.
```
```suggestion
`OpenAstronomy packaging guide`_ for packaging structure, which ensures a packaging
style consistent with the majority of existing Affiliated packages as well as large parts of
the greater scientific Python environment, in particular Numpy and Scipy.
However maintainers will find the pyOpenSci guide to be a useful reference, given it provides
community-wide guidelines and is targeted to those newer to packaging.
Editors and reviewer shall not let a particular choice of packaging influence their rating of
a package.
```
some rationale and clarification based on the discussion on @Cadair's comments.

dhomeier replied:
Actually this may warrant some further clarification in that we will require _some_ sort of packaging structure.
E.g. we have just recently included having a `CITATION` file in the requirements for new Affiliated packages, and should continue to ask for something with that function, though we would not demand it is called exactly that, or where in the repository it is sitting.

lwasser replied:
@dhomeier i really appreciate these edits but i'd love to chat a bit more about the language. 

> for packaging structure, which ensures a packaging style consistent with the majority of existing Affiliated  packages as well as large parts of the greater scientific Python environment, in particular Numpy and Scipy.

as i really worry it will confuse users. It is true that all of the core packages - numpy, scipy etc. use a flat layout! i actually did a survey of the ecosystem. but it's not true that the greater scientific Python ecosystem embraces it. 

There are many packages using the src/ layout now. in fact the entire scikit-hep community (and [this is now in the scientific-python.org packaging guide which really represents those core packages](https://learn.scientific-python.org/development/tutorials/packaging/)). As such what do you think about loosening up this language a bit?  

Maybe something like 

> packaging
style consistent with the majority of existing astropy Affiliated packages as well as the core scientific Python packages such as Numpy and Scipy.  

instead of :
> as well as large parts of
the greater scientific Python environment, in particular Numpy and Scipy.

i just worry about our users getting confused. because scientific-python.org, pypa, setuptools etc all are promoting src/ layout in their docs at this point. so i'd rather see more relaxed language. i welcome your thought and really understand and appreciate where you are coming from.

dhomeier replied:
I am entirely open to rewording, and happy to leave it to native speakers to iron out the fine print.
Regarding content, I think from https://github.com/astropy/astropy-APEs/pull/87#discussion_r1360898262 there really is consensus that we don't want to force a package to change their existing and functional layout.
But I doubt there is as broad support for recommending the scientific-python.org guide over the Openastronomy one for packages just setting their structure up – think it really comes down to balancing potential ease of configuration against joining or staying in a familiar environment.
For me personally I'll be much more involved in working with the code of other Astropy-Affiliated packages or the core scientific Python system (agreed that is a much better description for Numpy, Scipy etc.) than dealing with the internals of pypa, setuptools or even the scikit family. But that will obviously vary a lot individually.

hamogu replied:
I suggest that we worry too much about the language here. This is an APE where we propose to change our process, it's not the help page that users will see in in the docs. 
We also all agree that different layouts are possible, so I suggest we lead with that upfront. Something like

> There are different standards in the Python community for how to lay out a package and we do not force any particular choice. However, we encourage to follow  ... openastronomy ... for consistency ...

lwasser replied:
this all sounds wonderful to me!! @dhomeier i hear you!! and @hamogu i like your suggested wording. and got it - it's just an APE! i'm just hoping we can use language throughout our scientific ecosystem that helps users understand there are options and it's ok if they have already selected an option if it is working for them. i just know ppl get confused from talking to them BUT like you said this APE isn't necessarily public facing documentation  ! i think we have consensus here.

dhomeier replied:
Changed some of the wording according to the first comments

pllim replied:
Okay, given that Derek has edited his suggestion above to account for the comments here, I am going to accept it. Thanks!

dhomeier reviewed (changes requested):
Agreeing with all the changes now made in response to @hamogu's round of comments.
I have not made any concrete suggestions on keeping the badge system alive, as it is perhaps up for discussion whether it is really important/useful enough to spend much effort in. Anyway probably should not block this APE.

lwasser commented:
> I am fully in favor of joining pyOpenSci and replacing our current process with the PyOpenSci process.
> 
> I'm also convinced that we can work out any small questions (e.g. do we recommend the OpenAstronomy packaging or pyopensci or both?).
> 
> However, I'm very surprised by how negative this APE describes our current process for affiliated packages - I'm even more surprised because both our current affiliated edotors are listed as co-authors. I've left specific suggestion in places, but the general tone in this APE is to bad-mouth our existing process and I don't think that's helpful. This is an APE for changing a process, and we should fairly describe what's happening - pyOpenSci has a lot to offer without us writing this to make Astropy look bad - because I think it's not. The concept of affilated packages - that's been in place for a decade - has been very successful for the growth of our ecosystem.

Friends - i hope that you all know that i hold your existing review process with high respect. it's time consuming to build a review process and you've been developing / running this community run process for a solid 10 years! so i just wanted to re-inforce the fact that i hope our partnership supports astropy. Perhaps by leveraging our processes and community it  will even create space for other work that you need to do as a community.

pllim commented:
Sorry for the delay!

@dhomeier , I think I addressed your comments, except for one that I do not understand, which I have replied above.

How is this APE looking now? Anymore changes needed?

dhomeier commented on APE24.rst:
```diff
@@ -0,0 +1,477 @@
+
+pyOpenSci already has an established software peer review process and it is community-driven.
+It is compatible with Astropy but it is up to us to decide if a package
+applied to pyOpenSci also qualifies as Affiliated.
```
It was just my understanding that packages would be accepted _by_ pyOpenSci to JOSS and into the pyOpenSci (ecosystem? package directory), but in the meantime I've also seen them referred to just as "pyOpenSci packages", as in https://github.com/astropy/astropy-project/issues/334#issuecomment-1551657063.
So it's just a matter of fine print.

pllim commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+
+pyOpenSci already has an established software peer review process and it is community-driven.
+It is compatible with Astropy but it is up to us to decide if a package
+applied to pyOpenSci also qualifies as Affiliated.
```
Thanks! I accepted your changes.

If the content is good now, can you please approve? If not, please tell me what else needs changing.

dhomeier commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+   These new guidelines need to be compatible with
+   `pre-APE 24 Affiliated Package review guidelines`_ and
+   `pre-APE 24 Affiliated Package review process, with templates`_.
+   At the same time, we should link to more modern
+   practices over at pyOpenSci and `Scientific Python`_, in addition
+   to the "old ways" in Astropy and OpenAstronomy.
```
```suggestion
   At the same time, we should link to the actively developed
   practices over at pyOpenSci and `Scientific Python`_, in addition
   to the already implemented ones in Astropy and OpenAstronomy.
```
I think this captures it better than "more modern" (although OpenAstronomy is also still being actively developed),
but perhaps also reminds us that not everything can be substituted at once (e.g. the pyOS guide does not have a section on testing yet).

lwasser replied:
hey @pllim @dhomeier 

1. could we please link to our packaging guide here ??

_practices over at pyOpenSci
https://www.pyopensci.org/python-package-guide/_

2. I wonder if we want to be more explicit and say 
At the same time, we should link to the actively developed
   packaging guides developed by the pyOpenSci (link to our giude) and `Scientific Python`_, (link to scipy / henry's guide) communities in addition
   to the already implemented guidelines found in Astropy and OpenAstronomy (link to the open astronomy).

 I would argue that the practices in both our guide and henry's at scientific python are more modern leaning. I think henry is literally updating that guide when new peps and tools come out as he is so invested in and knowledgable about that space!!

However, for me i don't personally have a preference for modern vs maintained in this APE and welcome @dhomeier language!! it's good language and supports the important work those who developed the astro guide have done.   

i love the idea of adopting "newer" packaging approaches in our ecosystem slowly and steadily to support users. 

i hope that helps @pllim ping me again if it doesnt

hamogu replied:
I'm not specifically opposed to any of the changes / links suggested here. However, I feel that we are wasting our time - this is an APE that will allow the Astropy community to formally accept or decline this change. It's not a living document. Once accepted, it's archived. Sure, we can go back and look at it, but it's not meant to be up-to-date, and it's not the doc the package authors will look at. 

So, my suggestion is to stop fine-tuning this and just close and accept it. Instead, we should spend our efforts in palces that people will actually look at when they propose and look up accepted pacakges.

lwasser replied:
i will go ahead and approve on my end y 'all. @hamogu my apologies i didn't understand exactly how this APE will be used /viewed.

dhomeier replied:
I have added the requested changes/links to packaging guides above.

Admittedly I was not sure about the binding power of the APE either, but in light of the comments from @Cadair and others about describing our current process in an overly negative way, just wanted to avoid framing things here again in a "modern" vs. "outdated" or "obsolete" picture.

dhomeier commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+  use to rebuild our own `Astropy Affiliated Packages`_ listing.
+  This way, the packages would be listed both over at pyOpenSci and at Astropy
+  websites.
+* An Astropy-specific page on the pyOpenSci website that would be a dedicated
+  link for only Affiliated packages. We could then include some other
+  information about the Astropy ecosystem and branding (logo, colors, etc).
```
Looking back at https://github.com/astropy/astropy-project/issues/334#issuecomment-1551657063 it is perhaps not clear yet how this representation would look for the currently existing (not pyOS-approved) Affiliated packages – would they just remain in our listing, maybe separate from the new feed (which may be technically inevitable anyway), or would there be also another link back from the pyOS page?
In principle Affiliated packages could be in one of 3-4 states in that phase:
1. Accepted under the new system (first-time or re-reviewed)
2. Previously accepted and not re-reviewed
3. Re-reviewed and sunsetted
4. [re-reviewed and found still conforming to old Astropy standards, but not to pyOpenSci ones]

pllim replied:
> it is perhaps not clear yet how this representation would look for the currently existing (not pyOS-approved) Affiliated packages

All the existing packages would be in a static "legacy" page (basically the current Affiliated page but we would stop updating it going forward and slap a giant "legacy" note on top).

I suspect they would have to "re-apply" via the new process to be listed in the pyOpenSci/Astropy page, especially if they were accepted more than 2 years ago. 

I tried to explain this in the `Implementation` section on Step 3. But as you can see, I am also not very sure about it.

Do you think we need a new section just for this? Do you have a better idea on how to do this?

dhomeier reviewed (commented):
Just some details found on the final read, but this should be coming to completion now.

lwasser reviewed (approved):
i think this looks good y'all. please let me know how else i can support all of the work here.  @pllim thank you for pushing this through and keeping us on track! i know herding cats is a significant amount of work! 🐱

pllim commented:
Thanks, everyone! I have accepted Derek's updated suggestion based on other feedback.

I am hoping that @WilliamJamieson can still have a look since he is the other Astropy Editor. But I guess this has been open long enough that CoCo can proceed to a final decision if William cannot get to this by the next CoCo meeting on 2023-12-18?

pllim commented:
CoCo met today (2023-12-18) and discussed this. Out of 5, 2 abstained because they are also authors of this APE, 2 approved, and 1 absent.

lwasser commented:
> CoCo met today (2023-12-18) and discussed this. Out of 5, 2 abstained because they are also authors of this APE, 2 approved, and 1 absent.

amazing news @pllim !!

eteq commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+
+status: Discussion
+
+.. NOTE: Most of this stuff is from https://docs.google.com/document/d/1hIl3y6nrcepv-tKKQLVB4u0a3fXACQ1Qrg0Y6VrJykg/view
```
```suggestion
.. note: This APE is derived from brainstorming and discussions in https://docs.google.com/document/d/1hIl3y6nrcepv-tKKQLVB4u0a3fXACQ1Qrg0Y6VrJykg/view
```

eteq commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+========
+
+Since the early days of the Astropy Project, we had recognized the need
+to accommodate and developed an internal process for acknowledging 
```
```suggestion
to accommodate and develop an internal process for acknowledging 
```

dhomeier replied:
Was a bit unclear what was originally meant, but I suppose I was also a little bit unclear, possibly omitting a (non-Oxford) comma in my suggestion https://github.com/astropy/astropy-APEs/pull/87#discussion_r1388757260 – my understanding being that the internal process has been developed since. It's both in the past now anyways....

eteq commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+`pyOpenSci software peer review <https://www.pyopensci.org/software-peer-review/about/intro.html>`_
+(hereafter, pyOpenSci).
+Between the two, while JOSS has existed longer, pyOpenSci is more closely aligned
+with what we already do in Affiliated. This APE overhauls our existing Affiliated
```
```suggestion
with what we already do with Astropy Affiliated Packages. This APE overhauls our existing Affiliated
```

eteq commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+Detailed description
+====================
+
+.. This section describes the need for the APE.  It should describe the existing
+.. problem that it is trying to solve and why this APE makes the situation better.
+.. It should include examples of how the new functionality would be used and
+.. perhaps some use cases.
```
```suggestion
```

eteq commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+Will the new process still foster a sense of community in Astropy?
+------------------------------------------------------------------
+
+Yes, we think so! We will still keep the `Astropy Affiliated Packages`_
+page, but now instead of hosting the content ourselves, we grab an RSS/XML feed
+from pyOpenSci for the packages marked as Affiliated. To keep the history,
+we will move the existing listing to a "legacy affiliation" page and keep it static.
```
A strict reading of this is that it applies only to affiliated and not coordinates packages.  Is that intended?  (if so it should be more explicit I think)

pllim replied:
Correct, because so far (even before this APE), decision on whether a package becomes Coordinated or not does not go through the Affiliated review process. Lately, we had promoted Affiliated packages to Coordinates (e.g., specreduce, specutils) but Editors did not have any say in that, so Coordinated is not covered by this APE.

How do you wish this to be stated explicitly?

eteq replied:
How about this?

```suggestion
Yes, we think so! We will still keep the `Astropy Affiliated Packages`_
page, but now instead of hosting the content ourselves, we grab an RSS/XML feed
from pyOpenSci for the packages marked as Affiliated. To keep the history,
we will move the existing listing to a "legacy affiliation" page and keep it static.

Coordinated packages are separate from this due to their tighter integration with
the Astropy project's team structure, so they will remain listed on this page without
a connection to pyOpenSci.
```

eteq commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+Will we no longer need Editors?
+-------------------------------
+
+We still need them! However, instead of managing the whole process themselves,
+they will now be part of the `pyOpenSci Editorial Board`_ and perform their
+duties within the pyOpenSci process as laid out in this APE.
```
This means the role will remain in the astropy roles list, is that correct?  Just also listed in the pyOpenSci process?

(A roles change technically requires additional steps, so I just want it to be explicit if so, but my read on this is that it is *not* the case.  Just wanting clarification.)

pllim replied:
Same role, different tasks, is how I view it. We definitely still need our own Editors but they would work within pyOpenSci process as laid out in this APE.

pllim replied:
So in practice, this role stays on Team page. pyOpenSci will have these same people listed on their Editors page and maybe even add Astropy badge to their profiles (@lwasser , please confirm).

eteq replied:
Perfect!  All good there then.

eteq commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+Branches and pull requests
+==========================
+
+.. Any pull requests or development branches containing work on this APE should be
+.. linked to from here.  (An APE does not need to be implemented in a single pull
+.. request if it makes sense to implement it in discrete phases). If no code is yet
+.. implemented, just put "N/A"
```
```suggestion
```

eteq commented on APE24.rst:
```diff
@@ -0,0 +1,507 @@
+Implementation
+==============
+
+.. This section lists the major steps required to implement the APE.  Where
+.. possible, it should be noted where one step is dependent on another, and which
+.. steps may be optionally omitted.  Where it makes sense, each  step should
+.. include a link related pull requests as the implementation progresses.
```
```suggestion
```

eteq reviewed (changes requested):
I am happy with this idea overall, so the one "absent" vote from before can be changed to a yes!

However, I highlighted several minor typos and stylistic things and am marking this "request changes" on that basis. These are minor enough that once they are taken care of/accepted I am happy to finish the merging process myself.

I also have one larger question that I'm sure we talked about so I'm confident it has been addressed but I couldn't find an explcit statement in this text: are we *guaranteed*, say at least one editor seat on pyopensci?  I think it would be good to say that, not so much because I'm worried pyopensci wouldn't do it, but as a way to ensure we within Astropy stay plugged in.

Basically, I want to ensure we have a role that the CoCo feels they should try to keep at least one community member in so that we don't end up saying "well pyopensci has got it, so we can let that lapse". (Partly just because that doesn't seem fair to @lwasser et al!)

lwasser commented:
> I also have one larger question that I'm sure we talked about so I'm confident it has been addressed but I couldn't find an explcit statement in this text: are we _guaranteed_, say at least one editor seat on pyopensci? I think it would be good to say that, not so much because I'm worried pyopensci wouldn't do it, but as a way to ensure we within Astropy stay plugged in.
> 
> Basically, I want to ensure we have a role that the CoCo feels they should try to keep at least one community member in so that we don't end up saying "well pyopensci has got it, so we can let that lapse". (Partly just because that doesn't seem fair to @lwasser et al!)

@eteq absolutely. I'm so glad you brought that up! 

I think it is absolutely critical that astropy have atleast 1 person on the pyOpenSci editorial board and that we ensure that we also stay in touch / communicate. We could even consider a checkin once a year or at whatever cadence makes sense to ensure everyone is on the same page!!

eteq commented:
Alright, I'm happy with this, as is the rest of the coco.  Merging!

eteq commented:
Oops, I was supposed to do the decision rationale, etc in this PR but I'll just do it directly in main instead.  My bad.

pllim commented:
Thanks, all!

eteq commented:
Not also this was renumbered to APE22 in 6964140

pllim commented:
Wait... why is the APE number changed?

pllim commented:
APE 22 is already claimed but just not merged:

* https://github.com/astropy/astropy-APEs/pull/85

Any chance you can change this back to original APE 24 to avoid confusion?

pllim commented:
Also there is a typo in the rationale: `commitee` -> `committee`

eteq commented:
@lwasser - I think I found the right ORCiD for you in https://doi.org/10.5281/zenodo.10581891, but let me know if any of that is wrong and I can fix the metadata!

eteq commented:
> Any chance you can change this back to original APE 24 to avoid confusion?

The procedure in  the README covers this @pllim - the numbering is in *merging* order.  the numbering prior to merging is to be thought of as preliminary.

(The backstory is that we were struggling with deciding what to do here, and then we wrote it into the instructions and no one seemed to mind... So if we want to change that it's just a matter of changing the README instructions.  I don't have a terribly strong opinion other than "we should make sure the instructions are explicit about it and follow whatever they say at any given time")

pllim commented:
🤪

dhomeier commented:
I suppose the final renaming is only done right in the merge commit then – any special tricks how to do this?
